.github/workflows/deployment-pr.yml

```diff
@@ -0,0 +1,152 @@
+          deployments: ${{ steps.current-deployments.outputs.list }}
+          prs: ${{ steps.deployment-candidates.outputs.list }}
+        run: |
+          TO_INSTALL=$(jq --null-input --argjson prs '${{ env.prs }}' --argjson deployments '${{ env.deployments }}' \
+            '$prs|map(select([{name:.name,sha:.sha}]|inside($deployments)|not))')
+          echo "Will install the following candidates, because they either aren't deployed or their deployment is out of date: $TO_INSTALL"
+          echo "list=$TO_INSTALL" | tr -d "\n" >> $GITHUB_OUTPUT
+
+  uninstall-old-deployment:
+    name: Uninstall old deployment
+    runs-on: ubuntu-latest
+    needs:
+      - find-prs-to-deploy
+    if: fromJSON(needs.find-prs-to-deploy.outputs.to-uninstall)[0] != null
+    strategy:
+      fail-fast: false
+      matrix:
+        deployment-name: ${{ fromJSON(needs.find-prs-to-deploy.outputs.to-uninstall) }}
+    steps:
+
+      - name: Uninstall helm release
+        run: |
+          # Setup authentication
+          mkdir ~/.kube && echo '${{ secrets.KUBECONFIG }}' > ~/.kube/config && chmod go-r ~/.kube/config
+          # Uninstall the release
+          helm delete ecamp3-${{ matrix.deployment-name }}
+
+      - name: Remove GitHub deployment
+        uses: bobheadxi/deployments@v1.4.0
+        with:
+          step: deactivate-env
+          token: ${{ secrets.REPO_ACCESS_TOKEN }}
+          env: ${{ matrix.deployment-name }}
+
+      - name: Find former PR number based on deployment name
+        id: pr-number
+        env:
+          name: ${{ matrix.deployment-name }}
+          regex: '^pr([0-9]+)$'
+        run: |
+          [[ $name =~ $regex ]]
+          echo number=${BASH_REMATCH[1]} >> $GITHUB_OUTPUT
+
+      - name: Comment about uninstallation on PR
+        uses: thollander/actions-comment-pull-request@v2
+        if: steps.pr-number.outputs.number
+        with:
+          pr_number: ${{ steps.pr-number.outputs.number }}
+          message: |
+            ### <span aria-hidden="true">⛔</span> Feature branch deployment currently inactive.
+
+
+            If the PR is still open, you can add the `deploy!` label to this PR to trigger a feature branch deployment.
+          comment_tag: feature-branch-deployment-status
+          create_if_not_exists: false
+
+  build-and-push:
+    name: Build and push docker images  
+    needs: find-prs-to-deploy
+    if: fromJSON(needs.find-prs-to-deploy.outputs.to-deploy)[0] != null
+    uses: ./.github/workflows/reusable-build-and-push.yml
+    strategy:
+      fail-fast: false
+      matrix:
+        deployment: ${{ fromJSON(needs.find-prs-to-deploy.outputs.to-deploy) }}
+    with:
+      sha: ${{ matrix.deployment.sha }}
+    secrets:
+      DOCKER_HUB_USERNAME: ${{ vars.DOCKER_HUB_USERNAME }}
+      DOCKER_HUB_PASSWORD: ${{ secrets.DOCKER_HUB_PASSWORD }}
+
+  upgrade-or-install-deployment:
+    name: Upgrade or install deployment
+    needs:
+      - find-prs-to-deploy
+      - build-and-push
```

Comment:
Let me know if you tried and failed. I'm pretty sure it should work, but if not, I have no other complaints and would give my approval.

Reply:
Ah I remembered again why I opted for reusable workflow and not for an action.

Reusable workflows can be [called with `secrets: inherit`](https://github.com/ecamp/ecamp3/pull/3368/files#diff-d5240bbe3dac581ce22b64e9044cde8e47cbddc138846ff38f84d895c5aaf16aR151). In this case the sub-workflow has access to the secrets from the parent workflow. As the deployment step uses quite a bunch of secrets and there's a high chance, more secrets will be added, this is very convenient.

For actions this is not possible (see open issue https://github.com/actions/toolkit/issues/1168). If we want to switch to actions, the disadvantage is that each secret need be passed explicitly to the deployment action.

Reply:
I see. Then let's leave it at the reusable workflow. If we start running the PR deployments on actual events instead of cron schedule, this problem will be even less pronounced.